packages/template/src/lib/stack-app/apps/implementations/client-app-impl.ts

```diff
@@ -437,17 +523,137 @@ export class _StackClientAppImplIncomplete<HasTokenStore extends boolean, Projec
+  }
+  private _deleteLegacyBrowserRefreshCookies() {
+    deleteCookieClient(this._refreshTokenCookieName);
+    deleteCookieClient("stack-refresh");
+    const domain = this._getBrowserBaseDomain();
+    if (domain) {
+      deleteCookieClient(this._refreshTokenCookieName, { domain });
+      deleteCookieClient("stack-refresh", { domain });
+    }
+  }
+  private async _deleteLegacyServerRefreshCookies() {
+    const operations: Promise<unknown>[] = [
+      setOrDeleteCookie(this._refreshTokenCookieName, null, { noOpIfServerComponent: true }),
+      setOrDeleteCookie("stack-refresh", null, { noOpIfServerComponent: true }),
+    ];
+    const domain = await this._getServerBaseDomain();
+    if (domain) {
+      operations.push(
+        setOrDeleteCookie(this._refreshTokenCookieName, null, { domain, noOpIfServerComponent: true }),
+        setOrDeleteCookie("stack-refresh", null, { domain, noOpIfServerComponent: true }),
+      );
+    }
+    await Promise.all(operations);
+  }
+  private _queueCustomRefreshCookieUpdate(refreshToken: string | null, updatedAt: number | null, context: "browser" | "server") {
+    runAsynchronously(async () => {
+      const domain = await this._fetchTrustedParentDomain();
+      const previousDomain = this._lastCustomRefreshCookieDomain;
+      const setOrDeleteCookieWrapper = context === "browser"
+        ? async (...args: Parameters<typeof setOrDeleteCookieClient>) => setOrDeleteCookieClient(...args)
+        : setOrDeleteCookie;
+
+      const deleteCookie = async (targetDomain: string) => {
+        const name = this._getCustomRefreshCookieName(targetDomain);
+        await setOrDeleteCookieWrapper(name, null, { domain: targetDomain, noOpIfServerComponent: true });
+      };
+      const setCookie = async (targetDomain: string, value: string) => {
+        const name = this._getCustomRefreshCookieName(targetDomain);
+        await setOrDeleteCookieWrapper(name, value, { maxAge: 60 * 60 * 24 * 365, domain: targetDomain, noOpIfServerComponent: true });
+      };
+
+      if (!domain) {
+        if (previousDomain) {
+          await deleteCookie(previousDomain);
+          this._lastCustomRefreshCookieDomain = null;
+        }
+        return;
+      }
+
+      if (previousDomain && previousDomain !== domain) {
+        await deleteCookie(previousDomain);
+      }
+
+      if (refreshToken && updatedAt !== null) {
+        if (this._lastCustomRefreshCookieUpdatedAt !== null && updatedAt < this._lastCustomRefreshCookieUpdatedAt) {
+          // Ignore stale writes
+          return;
+        }
+        const value = this._formatRefreshCookieValue(refreshToken, updatedAt);
+        await setCookie(domain, value);
+        this._lastCustomRefreshCookieDomain = domain;
+        this._lastCustomRefreshCookieUpdatedAt = updatedAt;
+      } else {
+        await deleteCookie(domain);
+        this._lastCustomRefreshCookieDomain = null;
+        this._lastCustomRefreshCookieUpdatedAt = null;
+      }
```

Comment:
if we get rid of the delete in here that cleans up a lot:


```suggestion
        if (this._lastCustomRefreshCookieUpdatedAt !== null && updatedAt < this._lastCustomRefreshCookieUpdatedAt) {
          // Ignore stale writes
          return;
        }
        const value = this._formatRefreshCookieValue(refreshToken, updatedAt);
        await setCookie(domain, value);
        this._lastCustomRefreshCookieDomain = domain;
        this._lastCustomRefreshCookieUpdatedAt = updatedAt;
```